File: app/src/main/java/com/icerockdev/babenko/ui/BasePresenter.java
package com.icerockdev.babenko.ui;

/**
 * Created by Roman Babenko on 06/05/17.
 */

public abstract class BasePresenter<View> {
    private View mView;

    public View getView() {
        return mView;
    }

    public void attachView(View view) {
        mView = view;
    }

    public void detachView() {
        mView = null;
    }
}

File: app/src/main/java/com/icerockdev/babenko/ui/base/BasePresenter.kt
package com.icerockdev.babenko.ui.base

import com.arellomobile.mvp.MvpPresenter
import com.arellomobile.mvp.MvpView

/**
 * Created by Roman Babenko on 06/05/17.
 */

abstract class BasePresenter<View : MvpView> : MvpPresenter<View>()
